cli/compiler.rs

```diff
@@ -1,22 +1,87 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 use crate::isolate_state::IsolateState;
+use crate::isolate_state::{IsolateStateContainer, WorkerChannels};
 use crate::msg;
-use crate::permissions::{DenoPermissions, PermissionAccessor};
+use crate::ops;
 use crate::resources;
 use crate::resources::Resource;
 use crate::resources::ResourceId;
 use crate::startup_data;
 use crate::workers;
+use crate::workers::WorkerBehavior;
+use deno_core::deno_buf;
+use deno_core::deno_mod;
+use deno_core::Behavior;
 use deno_core::Buf;
+use deno_core::Op;
+use deno_core::StartupData;
 use futures::Future;
 use serde_json;
 use std::str;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
 use std::sync::Mutex;
 
 lazy_static! {
   static ref C_RID: Mutex<Option<ResourceId>> = Mutex::new(None);
 }
 
+pub struct CompilerBehavior {
+  pub state: Arc<IsolateState>,
+}
+
+impl CompilerBehavior {
+  pub fn new(state: Arc<IsolateState>) -> Self {
```

Comment:
It seems the CompilerBehavior is almost the same as CliBehavior except that it has a restricted set of ops?

Could this be made simpler by just having a boolean argument to `CliBehavior::new()` that says if it should have restricted ops or not? (It does mean a runtime check at dispatch, but that's cheap enough.)

Reply:
If you avoid having two types, it will simplify a lot of the types

Reply:
I did some more simplification, and removed the code duplication from `resolve`. My main reasons for the distinct `Behavior`s here are:

1. I don't expect there to be many changes to `Behavior`, so this code shouldn't have any reason to change often.
2. As you mentioned using a flag would be a extra compare operation.
3. When we implement web workers we might want a separate `Behavior` for them(#1866), and this will serve as a good example for whoever takes that on in the future. It will also ensure that the the infrastructure I created for having separate Behaviors will be maintained.

Reply:
SGTM